biochemist-python/chapters/Linear_Regression.ipynb

```diff
@@ -0,0 +1,359 @@
+    "import os\n",
+    "import numpy as np\n",
+    "import pandas as pd\n",
+    "import scipy as sp"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Importing data with pandas\n",
+    "In this lesson, you will continue to work with `os` and `pandas` to import your data. The exercises in this chapter represent a very straightforward case, but if you look at the code, you can see that it could be applied to much more complex situations. When using pandas, people often append \\_df to the end of a variable name as a reminder that this is a dataframe, so the dataframe that contains our results might be called results_df. The code in the next cell will enable you to create the dataframe for this lesson. It should be mentioned that DataFrame is a reserved word when working with pandas dataframes, so you should not use that word as a variable name."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "protein_file = os.path.join('data', 'protein_assay.csv') # gives a path to your csv file\n",
+    "results_df = pd.read_csv(protein_file) # use pandas to read the csv file into a dataframe\n",
+    "results_df  # display your dataframe"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "In a dataframe, you can refer to the data in a column just by the column header. These are strings, so you need to put single quotes around them when you use them. In pandas, the terms \"series\" is often used to refer to the data in a single column in a dataframe. So we are going to use the two column headers to define the data for our linear regression. "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "xdata = results_df['mass (ug)']   # Setting the x values  \n",
+    "ydata = results_df['A-595']  # setting the y values    \n",
+    "print(xdata,ydata) # checking to make sure everything is in place"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Linear Regression with SciPy\n",
+    "Now that the data are in place, we simply need to import the stats module from SciPy to perform the linear regression. \n",
+    "\n",
+    "Notice the the `linregress` function in scipy.stats accepts the two data series and actually generates five values: slope, intercept, r-value, p-value, and standard error. In the next cell all of those variables are assigned in a single command. Then the results are presented with a series of print statements."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "scrolled": true
+   },
+   "outputs": [],
+   "source": [
+    "from scipy import stats\n",
+    "slope, intercept, r_value, p_value, std_err = stats.linregress(xdata, ydata)\n",
+    "print(\"Slope = \", slope, \"/microgram/mL\", sep = \"\")\n",
```

Comment:
Thanks for pointing this out. I have been using my own, somewhat quirky approach to protein calibration curves for years. I checked the literature and the manuals and they all use protein concentration, rather than protein mass. I changed the units and the markdown to reflect protein concentration for the standards and the samples.